src/m3ninx/index/segment/fst/fst_terms_iterator.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fst
+
+import (
+	sgmt "github.com/m3db/m3/src/m3ninx/index/segment"
+	xerrors "github.com/m3db/m3x/errors"
+
+	"github.com/couchbase/vellum"
+)
+
+type newFstTermsIterOpts struct {
+	opts        Options
+	fst         *vellum.FST
+	finalizeFST bool
+}
+
+func (o *newFstTermsIterOpts) Close() error {
+	if o.finalizeFST {
+		return o.fst.Close()
+	}
+	return nil
+}
+
+func newFSTTermsIter(opts newFstTermsIterOpts) *fstTermsIter {
```

Comment:
nit: the capitalization of fst differs between the function and argument, `newFSTTermsIter` vs. `newFstTermsIterOpts`

Reply:
cheers, fixed in next patch